save-preprocessor/src/main/kotlin/org/cqfn/save/preprocessor/controllers/DownloadProject.kt

```diff
@@ -35,26 +38,39 @@ class DownloadProject {
     fun upload(@RequestBody gitRepository: GitRepository): Response {
         val urlHash = gitRepository.url.hashCode()
         val tmpDir = File("$volumes/$urlHash")
-        tmpDir.deleteRecursively()
+        if (tmpDir.exists()) {
+            tmpDir.deleteRecursively()
+            log.info("For ${gitRepository.url} repository: dir $urlHash was deleted")
+        }
+        tmpDir.mkdirs()
         log.info("For ${gitRepository.url} repository: dir $urlHash was created")
-        val user = if (gitRepository.username != null && gitRepository.password != null) {
+        val userCredentials = if (gitRepository.username != null && gitRepository.password != null) {
             UsernamePasswordCredentialsProvider(gitRepository.username, gitRepository.password)
         } else {
             CredentialsProvider.getDefault()
         }
         try {
             Git.cloneRepository()
                 .setURI(gitRepository.url)
-                .setCredentialsProvider(user)
+                .setCredentialsProvider(userCredentials)
                 .setDirectory(tmpDir)
                 .call().use {
                     log.info("Repository cloned: ${gitRepository.url}")
                     // TODO post request to orchestrator
                     return Mono.just(ResponseEntity("Cloned", HttpStatus.ACCEPTED))
                 }
+        } catch (invalidRemoteException: InvalidRemoteException) {
+            log.warn("Invalidate remote exception while cloning ${gitRepository.url} repository", invalidRemoteException)
+            return Mono.just(ResponseEntity(invalidRemoteException.stackTraceToString(), HttpStatus.BAD_GATEWAY))
```

Comment:
I think all jgit-specific exceptions should return BAD_REQUEST, because it means that data supplied from user is incorrect.

Reply:
Ok, fixed